examples/mcp-client/src/server.ts

```diff
@@ -58,12 +56,22 @@ export class MyAgent extends Agent<Env, State> {
 
   async addMcpServer(url: string): Promise<string> {
     console.log(`Registering server: ${url}`);
-    const { id, authUrl } = await this.mcp.connect(url);
+    const authProvider = new DurableObjectOAuthClientProvider(
+      this.ctx.storage,
+      this.name,
+      `${this.env.HOST}/agents/my-agent/${this.name}/callback`
+    );
+    const { id, authUrl } = await this.mcp.connect(url, {
+      transport: { authProvider },
+    });
     this.setServerState(id, {
       url,
       authUrl,
       state: this.mcp.mcpConnections[id].connectionState,
     });
+    if (this.mcp.mcpConnections[id].connectionState === "ready") {
+      await this.refreshServerData();
+    }
```

Comment:
what does this do? should it be running more often?

Reply:
It just refreshes the agent's state for the tools, prompts, and resources which get sent to the client. It should be sufficient to just update these upon adding a new server or authenticating to a server.

In the future I'd probably trigger this function on `listChanged`.